### DISCUSSION TITLE
Saving as a file is ok, but outputing to web page is corrupted

### DISCUSSION THREAD

**Discussion by qpzmp** [ImageMagick/ImageMagick#3103] [Help]
I am using ImageMagik on my Debian server (apt-get install php5.6-imagick).

As expected, this bit of code reads in a picture and saves a copy elsewhere:

**$canvas = new Imagick( "/var/www/websites/DisplayScreen/Slideshow Photos/20161016_151555.jpg" );
$canvas->writeImage("/var/www/websites/DisplayScreen/20161016_151555.jpg");**

When I change the code to:

**$canvas = new Imagick( "/var/www/websites/DisplayScreen/Slideshow Photos/20161016_151555.jpg" );
header( "Content-Type: image/jpg" );
echo $canvas;**

and call it from a web page, a file that is a similar size to the original is delevered but is not recognised by the browser (or Paint Shop Pro). The output image seems to have been corrupted.

I have tried **header( "Content-Type: image/jpeg" );** and **echo $canvas->getImageBlob();** as varients to the above, but get the same result.

All help gratefully received...

**Comment by urban-warrior** [comment]
See https://www.geeksforgeeks.org/php-imagick-displayimage-function/.

**Reply by qpzmp** [reply]
Thanks for your prompt reply.

When I try the geeksforgeeks exact code I get **Fatal error: Uncaught exception 'ImagickException' with message 'unable to open X server `5'**

Can you explain what this function does? There is no explanation in the manual or on geeksforgeeks.

**Comment by urban-warrior** [comment]
It looks like the displayImage() function wants to display to an X windows server rather than a web page.  However, that page has an example for the web that utilizes the getImageBlob() call.  Note, we did not write Imagick, nor do we support it.  You might want to post to the Imagick developers or go to StackOverflow to get help.
